src/main/java/org/jenkinsci/plugins/docker/workflow/client/LaunchResult.java

```diff
@@ -46,22 +48,22 @@ LaunchResult setStatus(int status) {
         return this;
     }
 
-    public String getOut() {
+    public @CheckForNull String getOut() {
```

Comment:
Does not make sense. It should be `@Nonnull`, and the weird code in `launch` should be cleaned up.

```java
try {
    result.setStatus(procStarter.quiet(quiet).cmds(args).envs(launchEnv).stdout(out).stderr(err).join());
    return result;
} finally {
    try {
        result.setOut(out.toString());
        out.close();
    } finally {
        result.setErr(err.toString());
        err.close();
    }
}
```

can probably be simplified to

```java
result.setStatus(procStarter.quiet(quiet).cmds(args).envs(launchEnv).stdout(out).stderr(err).join());
result.setOut(out.toString());
result.setErr(err.toString());
return result;
```

(If `join` throws an exception, there is no point adjusting `result`. If it returns normally, we can assume stdio has been flushed. And a `ByteArrayOutputStream` need not be `close`d.)